igvc_utils/include/igvc_utils/NodeUtils.hpp

```diff
@@ -1,31 +1,131 @@
+  assert_with_default(node_namespace, variable, variable > 0, std::forward<T>(default_value), message.str());
+}
+
+template <class T>
+inline void assert_negative_with_default(const std::string &node_namespace, T &variable, T &&default_value,
+                                         const std::string &variable_name)
+{
+  std::ostringstream message;
+  message << "[" << node_namespace << "] " << variable_name << " (currently " << variable
+          << ") should be less than 0. Setting to default value of " << default_value;
+  assert_with_default(node_namespace, variable, variable < 0, std::forward<T>(default_value), message.str());
+}
+
+template <class T>
+inline void check_assertion_with_default(Assertion assertion, const std::string &node_namespace, T &variable,
+                                         T &&default_value, const std::string &variable_name)
+{
+  switch (assertion)
+  {
+    case Assertion::POSITIVE:
+      assert_positive_with_default(node_namespace, variable, std::forward<T>(default_value), variable_name);
+      break;
+    case Assertion::NEGATIVE:
+      assert_negative_with_default(node_namespace, variable, std::forward<T>(default_value), variable_name);
+      break;
+    case Assertion::NONE:
+      break;
+  }
+}
+
+template <class T>
+inline void assert_positive(const std::string &node_namespace, T &variable, const std::string &variable_name)
+{
+  ROS_ERROR_STREAM("[" << node_namespace << "] " << variable_name << " (currently " << variable
+                       << ") should be greater than 0. Exiting...");
+  ros::shutdown();
+}
+
+template <class T>
+inline void assert_negative(const std::string &node_namespace, T &variable, const std::string &variable_name)
+{
+  ROS_ERROR_STREAM("[" << node_namespace << "] " << variable_name << " (currently " << variable
+                       << ") should be less than 0. Exiting...");
+  ros::shutdown();
+}
+
+template <class T>
+inline void check_assertion(Assertion assertion, const std::string &node_namespace, T &variable,
+                            const std::string &variable_name)
+{
+  switch (assertion)
+  {
+    case Assertion::POSITIVE:
+      assert_positive(node_namespace, variable, variable_name);
+      break;
+    case Assertion::NEGATIVE:
+      assert_negative(node_namespace, variable, variable_name);
+      break;
+    case Assertion::NONE:
+      break;
+  }
+}
+
+template <class T>
+void param(const ros::NodeHandle &pNh, const std::string &param_name, T &param_val, T &&default_val,
+           Assertion assertion = Assertion::NONE)
 {
   if (!pNh.param(param_name, param_val, default_val))
   {
-    ROS_ERROR_STREAM("Missing parameter " << param_name << " from " << pNh.getNamespace()
-                                          << ". Continuing with default values " << default_val);
+    ROS_WARN_STREAM("[" << pNh.getNamespace() << "] Missing parameter " << param_name
```

Comment:
Compiler error thrown here when `default_val` doesn't have an overloaded `<<` operator

Reply:
Ddin't fix this, but I added a template specialization for `std::vector`, which should be good enough for now.